src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -72,6 +72,16 @@ module.exports = function<T, P, I, C>(config : HostConfig<T, P, I, C>) {
     var fn = workInProgress.type;
     var props = workInProgress.pendingProps;
 
+    // TODO: Disable this before release, since it is not part of the public API
+    // I use this for testing to compare the relative overhead of classes.
+    if (typeof fn.shouldComponentUpdate === 'function') {
+      if (workInProgress.memoizedProps !== null) {
```

Comment:
Can we swap the conditionals? fn.shouldComponentUpdate is probably slower than workInProgress.memoizedProps. Also, maybe use a crazy name like shouldComponentUpdateForFiberTestingOnly so no one gets confused?